frontend/src/features/contacts/contact/create/address/useAddressHelpers.ts

```diff
@@ -0,0 +1,58 @@
+import { SelectOption } from 'components/common/form';
+import * as API from 'constants/API';
+import { CountryCodes } from 'constants/countryCodes';
+import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
+import { Dictionary } from 'interfaces/Dictionary';
+import { useEffect, useMemo, useState } from 'react';
+
+const formLabelMap = new Map<string, Dictionary<string>>([
+  [
+    CountryCodes.Canada,
+    { province: 'Province', provincePlaceholder: 'Select province', postal: 'Postal Code' },
+  ],
+  [CountryCodes.US, { province: 'State', provincePlaceholder: 'Select state', postal: 'Zip Code' }],
+  [CountryCodes.Other, { province: '', provincePlaceholder: '', postal: 'Postal Code' }],
+]);
+
+/**
+ * Hook that provides several helpers to the Address component.
+ */
+export default function useAddressHelpers() {
```

Comment:
this makes sense, might as well group common logic like this to clean up the components that use this.

Reply:
yeah - I've been using "local hooks" to try to keep the component as lean as possible while moving logic off to the hook. I know this one might not be reused but keeps the component clean